### DISCUSSION TITLE
Technical question about bit 7 (mute voice 3) in $D418 and bit 2 in $D417

### DISCUSSION THREAD

**Discussion by Toddler-Boy** [libsidplayfp/libsidplayfp#86] [Q&A] [answered]
So what happens if bit 7 ($80) in $D418 is set (mute voice 3) but at the same time bit 2 of $D417 is set, which routes the voice through the filter. Does voice3 play through the filter after all, or does bit 7 in $D418 take precedence and mutes it?

**Comment by drfiemost** [comment]
It goes through the filter

https://github.com/libsidplayfp/libsidplayfp/blob/b0ceb7672325c1025b5ae7aa28cd4d5afc3d3f77/src/builders/residfp-builder/residfp/Filter8580.h#L357-L358

**Comment by Toddler-Boy** [comment]
Thank you :)

And what if the filter mode has no bits set (low-pass, band-pass, and high-pass are not set)? Are any voices routed through a "null" filter audible?

**Comment by drfiemost** [comment, marked as answer]
Ideally no, but there may be some small signal leak as the manufacturing is far from perfect.

**Comment by Toddler-Boy** [comment]
I’m just wondering because I’ve noticed a lot of tunes from Spiro2006 that have semi complex filter modulations, despite the filter being off 100% of the time. I thought it might be some weird trick to get a certain sound, but I can’t hear any filter movement either.

**Comment by drfiemost** [comment]
You can use [SIDDump](https://csdb.dk/release/?id=192079) to check what the tune is doing.

**Comment by Toddler-Boy** [comment]
Oh, I can clearly see what the tune is doing. I can see the filter movement as a visual graph, etc.

![image](https://github.com/libsidplayfp/libsidplayfp/assets/2496353/55c53ed8-ac9c-45f8-804d-2b458393f32b)

The filter section is grey, because none of the three voices is using it, but you can clearly see that the cutoff is being written to with sawtooth-like shapes, seemingly with intent. But with the filter not being used, you can never hear it. This is true for a lot of Spino2006 tunes and I just wondered if it's something perhaps more subtle. So I figured I'd ask, so see if there is a another way to utilize the filter that is not obvious to me.

I draw the filter section in grey when either none of the three voices have the filter flags set OR the filter mode has none of its three modes set (low-pass, band-pass, high-pass).

The tune in question is "Cosmic Energy (extended)", but this is true for a lot of Spino2006's tunes.

Here is an example from Vincenzo. Tune is called "Solar Flare".

![image](https://github.com/libsidplayfp/libsidplayfp/assets/2496353/1c627d13-27f2-4551-8503-868039ee7bcf)

As you can see, voice1 uses the filter, low-pass is on, and the cutoff and resonance are heavily modulated.

**Comment by drfiemost** [comment]
Oh, I see. Maybe it's just a leftover or (unlikely) a subtle effect on the backround noise. Need to check on a real SID.

**Comment by Toddler-Boy** [comment]
I’ve contacted him. He intentionally doesn’t use the filter because he’s afraid of sound differences between his 8580 and other users 6581.
